content/storage/docs/spark/instance-store.md

```diff
@@ -0,0 +1,146 @@
+
+**Example**
+
+```
+apiVersion: eksctl.io/v1alpha5
+kind: ClusterConfig
+
+metadata:
+  name: YOUR_CLUSTER_NAME
+  region: YOUR_REGION
+
+managedNodeGroups:
+  - name: ng-m5d-4xlarge
+    instanceType: m5d.4xlarge
+    desiredCapacity: 1
+    privateNetworking: true
+    subnets:
+      - YOUR_NG_SUBNET
+    preBootstrapCommands: # commands executed as root
+      - "IDX=1;for DEV in /dev/nvme[1-9]n1;do mkfs.xfs ${DEV}; mkdir -p /spark_data_${IDX}; echo ${DEV} /spark_data_${IDX} xfs defaults,noatime 1 2 >> /etc/fstab; IDX=$((${IDX} + 1)); done"
+      - "mount -a"
+      - "chown 999:1000 /spark_data_*"
+```
+
+
+In order to successfully use these volumes with Spark, it’s required to specify additional configurations to use these volumes during shuffles operations. To use a volume as local storage, the volume’s name should starts with `spark-local-dir-`.
+
+Below an example configuration provided during the EMR on EKS job submission, that shows how to configure Spark to use 2 volumes as local storage for the job.
+
+**Spark Configurations**
+
+```
+{
+  "name": ....,
+  "virtualClusterId": ....,
+  "executionRoleArn": ....,
+  "releaseLabel": ....,
+  "jobDriver": ....,
+  "configurationOverrides": {
+    "applicationConfiguration": [
+      {
+        "classification": "spark-defaults",
+        "properties": {
+          "spark.kubernetes.executor.volumes.hostPath.spark-local-dir-1.mount.path": "/spark_data_1",
+          "spark.kubernetes.executor.volumes.hostPath.spark-local-dir-1.mount.readOnly": "false",
+          "spark.kubernetes.executor.volumes.hostPath.spark-local-dir-1.options.path": "/spark_data_1",
+          "spark.kubernetes.executor.volumes.hostPath.spark-local-dir-2.mount.path": "/spark_data_2",
+          "spark.kubernetes.executor.volumes.hostPath.spark-local-dir-2.mount.readOnly": "false",
+          "spark.kubernetes.executor.volumes.hostPath.spark-local-dir-2.options.path": "/spark_data_2"
+        }
+      }
+    ]
+  }
+}
+```
+
+
+Please note that for this approach it is required to specify the following configurations for each volume that you want to use. (IDX is a label to identify the volume mounted)
+
+```
+# Mount path on the host node
+spark.kubernetes.executor.volumes.hostPath.spark-local-dir-IDX.options.path
+
+# Mount path on the k8s pod
+spark.kubernetes.executor.volumes.hostPath.spark-local-dir-IDX.mount.path
+
+# (boolean) Should be defined as false to allow Spark to write in the path
+spark.kubernetes.executor.volumes.hostPath.spark-local-dir-IDX.mount.readOnly
+```
+
+
+**Benefits**
+
+* Possible to re-partition volumes across different Spark jobs (e.g. a node group containing instances with 2 instance volumes each, can dedicate one volume per Spark job)
```

Comment:
You can allocate dedicated resources of instance store volumes across your Spark jobs (For example, lets take a scenario where an EC2 instance has two instance store volumes. If you run two spark jobs on this node, you can dedicate one volume per Spark job)